de-lib/src/main/java/org/iplantc/de/client/services/DiskResourceServiceFacade.java

```diff
@@ -129,7 +131,7 @@ void moveContents(final Folder sourceFolder,
      * @param dest id of the destination folder.
      * @param callback service success/failure callback
      */
-    void importFromUrl(String url, DiskResource dest, AsyncCallback<String> callback);
+    void importFromUrl(String url, DiskResource dest, DECallback<String> callback);
```

Comment:
That's a good idea; we should be able to make the nginx return that automatically if we add a JSON body to the 502/503/504 response, and then we just need to make sure our error-handling code passes that info along. That'll probably need more investigation, and it's possible some of these endpoints should more gracefully respond when stuff is down anyway (e.g. the manifest endpoint)